src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php

```diff
@@ -170,6 +158,18 @@ public function getTypes($class, $property, array $context = []): ?array
             return $fromDefaultValue;
         }
 
+        if (\PHP_VERSION_ID >= 70400) {
```

Comment:
I think it'll make more sense to have precedence over constructor and default value extraction (the two if above), right?

Reply:
@ogizanagi yeah it might make more sense in the fact that if there is constructor args or default values as well as a php7.4 property type it might be more simple to just extract from the property type? I'm not sure it makes much of a difference though. Happy to move it if you like though.

Reply:
Looks like moving this block has been changing the previous behaviour quite a bit. 
https://github.com/symfony/symfony/issues/38416

Maybe you could have a look, if this could be reverted (while still fixing #38041)